frappe/public/js/frappe/form/footer/timeline.js

```diff
@@ -560,26 +560,38 @@ frappe.ui.form.Timeline = class Timeline {
 				return;
 			}
 
-			let data_import_link = frappe.utils.get_form_link(
-				'Data Import Beta',
-				data.data_import,
-				true,
-				__('via Data Import')
-			);
+			let updater_reference_link = null;
+
+			if (data.data_import) {
+				updater_reference_link = frappe.utils.get_form_link(
+					'Data Import Beta',
+					data.data_import,
+					true,
+					__('via Data Import')
+				);
+			} else if (!$.isEmptyObject(data.updater_reference)) {
+				let updater_reference = data.updater_reference;
+				updater_reference_link = frappe.utils.get_form_link(
+					updater_reference.doctype,
+					updater_reference.docname,
+					true,
+					__('via {0}', [updater_reference.doctype])
```

Comment:
this text should come from `updater_reference.label`, then data_import wont be a special case